### ISSUE TITLE
Error in rule run_das_tool - "there is no package called 'doMC'"

### ISSUE THREAD

**Issue by LLansing** [metagenome-atlas/atlas#496]
- [x] ** I got an error in the following rule(s):** `run_das_tool`

- [x] I checked the log files indicated indicated in the error message (and the cluster logs if submitted to a cluster)

Here is the relevant log output:

```
Error in rule run_das_tool:
    jobid: 4
    output: sample1/binning/DASTool/sample1_DASTool_summary.txt, sample1/binning/DASTool/sample1_metabat.eval, sample1/binning/DASTool/sample1_maxbin.eval, sample1/binning/DASTool/cluster_attribution.tsv
    log: sample1/logs/binning/DASTool.log (check log file(s) for error message)
    conda-env: /isilon/lethbridge-rdc/users/ortegapoloro/lance_test/metagenome_assembly_project/atlas/databases/conda_envs/3a9d745df06801029b0dc0bf9e808bfc
    shell:
         DAS_Tool --outputbasename sample1/binning/DASTool/sample1  --bins sample1/binning/DASTool/metabat.scaffolds2bin,sample1/binning/DASTool/maxbin.scaffolds2bin  --labels metabat,maxbin  --contigs sample1/sample1_contigs.fasta  --search_engine diamond  --proteins sample1/annotation/predicted_genes/sample1.faa  --write_bin_evals 1  --megabin_penalty 0.5 --duplicate_penalty 0.6  --threads 8  --debug  --score_threshold 0.5 &> sample1/logs/binning/DASTool.log  ; mv sample1/binning/DASTool/sample1_DASTool_scaffolds2bin.txt sample1/binning/DASTool/cluster_attribution.tsv &>> sample1/logs/binning/DASTool.log
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)
    cluster_jobid: 1596930

Error executing rule run_das_tool on cluster (jobid: 4, external: 1596930, jobscript: /isilon/lethbridge-rdc/users/ortegapoloro/lance_test/metagenome_assembly_project/atlas/.snakemake/tmp.ag32jxy1/snakejob.run_das_tool.4.sh). For error details see the cluster log and the log files of the involved rule(s).
[Mon Feb 14 16:23:54 2022]
Error in rule run_das_tool:
    jobid: 36
    output: sample2/binning/DASTool/sample2_DASTool_summary.txt, sample2/binning/DASTool/sample2_metabat.eval, sample2/binning/DASTool/sample2_maxbin.eval, sample2/binning/DASTool/cluster_attribution.tsv
    log: sample2/logs/binning/DASTool.log (check log file(s) for error message)
    conda-env: /isilon/lethbridge-rdc/users/ortegapoloro/lance_test/metagenome_assembly_project/atlas/databases/conda_envs/3a9d745df06801029b0dc0bf9e808bfc
    shell:
         DAS_Tool --outputbasename sample2/binning/DASTool/sample2  --bins sample2/binning/DASTool/metabat.scaffolds2bin,sample2/binning/DASTool/maxbin.scaffolds2bin  --labels metabat,maxbin  --contigs sample2/sample2_contigs.fasta  --search_engine diamond  --proteins sample2/annotation/predicted_genes/sample2.faa  --write_bin_evals 1  --megabin_penalty 0.5 --duplicate_penalty 0.6  --threads 8  --debug  --score_threshold 0.5 &> sample2/logs/binning/DASTool.log  ; mv sample2/binning/DASTool/sample2_DASTool_scaffolds2bin.txt sample2/binning/DASTool/cluster_attribution.tsv &>> sample2/logs/binning/DASTool.log
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)
    cluster_jobid: 1596929

Error executing rule run_das_tool on cluster (jobid: 36, external: 1596929, jobscript: /isilon/lethbridge-rdc/users/ortegapoloro/lance_test/metagenome_assembly_project/atlas/.snakemake/tmp.ag32jxy1/snakejob.run_das_tool.36.sh). For error details see the cluster log and the log files of the involved rule(s).
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message

```

The `DASTool.log` found at `sample2/logs/binning/` is about 450 lines of output that seems to show DAS_tool had some amount of success running, but log ends with the following lines:

```
...
Cleaning up...  [0s]
Total time = 4.138s
Reported 68 pairwise alignments, 68 HSPs.
68 queries aligned.
starting annotations of single copy cogs...
successfully finished
calculating contig lengths.
Error in library(x, character.only = TRUE) : 
  there is no package called ‘doMC’
Calls: lapply ... suppressPackageStartupMessages -> withCallingHandlers -> library
Execution halted
mv: cannot stat ‘sample2/binning/DASTool/sample2_DASTool_scaffolds2bin.txt’: No such file or directory
```

** Atlas version**
2.8.1

**Additional context**

Running Atlas on an cluster with an SGE job scheduler. I made a snakemake profile for SGE and it seems to be working fine (at least for all other jobs up until this point.

A similar error with DAStool was reported [here](https://github.com/jtamames/SqueezeMeta/issues/29) , but I'm not sure how to implement the correction that user mentions.

**Comment by SilasK** [comment]
Try `mamba install -p /isilon/lethbridge-rdc/users/ortegapoloro/lance_test/metagenome_assembly_project/atlas/databases/conda_envs/3a9d745df06801029b0dc0bf9e808bfc -c conda-forge r-domc`

or with conda if this doesn't work.

**Comment by SilasK** [comment]
Did it solve the issue?

**Comment by LLansing** [comment]
In both cases (mamba and conda) it says:
  All requested packages already installed

**Comment by SilasK** [comment]
I forwarded the issue to https://github.com/cmks/DAS_Tool/issues/75

As a fix, I recommend to not use DASTool. I suggest you to use either `'metabat'` or even better `'SemiBin'` as final binner in the config file. Semibin has shown a great improvement.

**Comment by cmks** [comment]
This looks looks like a R library path issue to me. Make sure that the library path used for the execution of DAS Tool matches the one of the installation. You can check the path using `.libPaths()` from inside R.

**Comment by cmks** [comment]
@LLansing I've recently released a new DAS Tool version (1.1.4) which does not require the doMC R package any more (https://github.com/cmks/DAS_Tool/releases/tag/1.1.4, https://anaconda.org/bioconda/das_tool). If doMC caused the issues on your system, you should consider updating.
